src/mgr/ClusterState.cc

```diff
@@ -72,7 +72,14 @@ void ClusterState::ingest_pgstats(ref_t<MPGStats> stats)
   std::lock_guard l(lock);
 
   const int from = stats->get_orig_source().num();
-  pending_inc.update_stat(from, std::move(stats->osd_stat));
+  auto is_in = with_osdmap([&](const OSDMap& osdmap) {
```

Comment:
i assume at this moment, the mgr has been updated with the latest osdmap, hence `ClusterState::notify_osdmap()` should have been called, and `pending_inc.osd_stat_updates` and `existing_pools` should have been updated to be in sync with osdmap. so i think we can just ignore the stats from osd which is out.

Reply:
@tchaikov Yes, we can do that. Pls review the new changes, thanks.

Reply:
```suggestion
  bool is_in = with_osdmap([from](const OSDMap& osdmap) {
```

for better readability.

Reply:
@tchaikov done.